lib/OpenQA/Schema/ResultSet/JobSettings.pm

```diff
@@ -68,4 +68,14 @@ sub query_for_settings ($self, $args) {
     return $self->search({-and => \@conds});
 }
 
+sub parent_worker_class ($self, $parent_job_id, $worker_class) {
+    my $result = $self->search({parent_job_id => $job_id, key => 'WORKER_CLASS'})->next;
```

Comment:
if we only want one row, it's better to tell that the database

```suggestion
    my $result = $self->search({parent_job_id => $job_id, key => 'WORKER_CLASS'}, {rows => 1})->single;
```
since there can be multiple WORKER_CLASS entries, we're still relying on the internal order of postgres here I guess.